File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/View/json/ItemsFactory.java
package com.spvessel.spacevil.View.json;

import com.spvessel.spacevil.*;
import com.spvessel.spacevil.Core.InterfaceBaseItem;
import com.spvessel.spacevil.Decorations.Style;
import com.spvessel.spacevil.Flags.*;

public class ItemsFactory {
    public static Object getObject(String name) {
        switch (name.charAt(0)) {
            case 'B': {
                if (name.equals("BlankItem")) {
                    return getBlankItem();
                } else if (name.equals("ButtonCore")) {
                    return getButtonCore();
                } else if (name.equals("ButtonToggle")) {
                    return getButtonToggle();
                }
                break;
            }
            case 'C': {
                if (name.equals("CheckBox")) {
                    return getCheckBox();
                } else if (name.equals("ComboBox")) {
                    return getComboBox();
                } else if (name.equals("ComboBoxDropDown")) {
                    return getComboBoxDropDown();
                    // } else if (name.equals("ContextMenu")) {
                    //     return getContextMenu();
                } else if (name.equals("CustomShape")) {
                    return getCustomShape();
                }
                break;
            }
            case 'E': {
                if (name.equals("Ellipse")) {
                    return getEllipse();
                }
                break;
            }
            case 'F': {
                // } else if (name.equals("FileSystemEntry")) {
                //     return getFileSystemEntry();
                // } else if (name.equals("FloatItem")) {
                //     return getFloatItem();
                if (name.equals("Frame")) {
                    return getFrame();
                } else if (name.equals("FreeArea")) {
                    return getFreeArea();
                }
                break;
            }
            case 'G': {
                if (name.equals("Graph")) {
                    return getGraph();
                } else if (name.equals("Grid")) {
                    return getGrid();
                }
                break;
            }
            case 'H': {
                if (name.equals("HorizontalScrollBar")) {
                    return getHorizontalScrollBar();
                } else if (name.equals("HorizontalSlider")) {
                    return getHorizontalSlider();
                } else if (name.equals("HorizontalSplitArea")) {
                    return getHorizontalSplitArea();
                } else if (name.equals("HorizontalStack")) {
                    return getHorizontalStack();
                }
                break;
            }
            case 'I': {
                if (name.equals("ImageItem")) {
                    return getImageItem();
                } else if (name.equals("Indicator")) {
                    return getIndicator();
                    // } else if (name.equals("InputBox")) {
                    //     return getInputBox();
                    // } else if (name.equals("InputDialog")) {
                    //     return getInputDialog();
                }
                break;
            }
            case 'L': {
                if (name.equals("Label")) {
                    return getLabel();
                } else if (name.equals("LinesContainer")) {
                    return getLinesContainer();
                } else if (name.equals("ListArea")) {
                    return getListArea();
                } else if (name.equals("ListBox")) {
                    return getListBox();
                } else if (name.equals("LoadingScreen")) {
                    return getLoadingScreen();
                }
                break;
            }
            case 'M': {
                if (name.equals("MenuItem")) {
                    return getMenuItem();
                } else if (name.equals("MessageBox")) {
                    return getMessageBox();
                } else if (name.equals("MessageItem")) {
                    return getMessageItem();
                }
                break;
            }
//            case 'O':  {
//                if (name.equals("OpenEntryBox")) {
//                    return getOpenEntryBox();
//                } else if (name.equals("OpenEntryDialog")) {
//                    return getOpenEntryDialog();
//                }
//                break;
//            }

            case 'P': {
                if (name.equals("PasswordLine")) {
                    return getPasswordLine();
                } else if (name.equals("PointsContainer")) {
                    return getPointsContainer();
                    // } else if (name.equals("PopUpMessage")) {
                    //     return getPopUpMessage();
                } else if (name.equals("ProgressBar")) {
                    return getProgressBar();
                }
                break;
            }
            case 'R': {
                if (name.equals("RadioButton")) {
                    return getRadioButton();
                } else if (name.equals("Rectangle")) {
                    return getRectangle();
                } else if (name.equals("ResizableItem")) {
                    return getResizableItem();
                }
                break;
            }
            case 'S': {
                if (name.equals("ScrollHandler")) {
                    return getScrollHandler();
                    // } else if (name.equals("SelectionItem")) {
                    //     return getSelectionItem();
                    // } else if (name.equals("SideArea")) {
                    //     return getSideArea();
                } else if (name.equals("SpinItem")) {
                    return getSpinItem();
                    // } else if (name.equals("SplitHolder")) {
                    //     return getSplitHolder();
                } else if (name.equals("Style") || name.equals("InnerStyle")) { //???
                    return getStyle();
                }
                break;
            }
            case 'T': {
                if (name.equals("Tab")) {
                    return getTab();
                } else if (name.equals("TabView")) {
                    return getTabView();
                } else if (name.equals("TextArea")) {
                    return getTextArea();
                } else if (name.equals("TextEdit")) {
                    return getTextEdit();
                } else if (name.equals("TextView")) {
                    return getTextView();
                } else if (name.equals("TitleBar")) {
                    return getTitleBar();
                    // } else if (name.equals("TreeItem")) {
                    //     return getTreeItem();
                } else if (name.equals("TreeView")) {
                    return getTreeView();
                } else if (name.equals("Triangle")) {
                    return getTriangle();
                }
                break;
            }
            case 'V': {
                if (name.equals("VerticalScrollBar")) {
                    return getVerticalScrollBar();
                } else if (name.equals("VerticalSlider")) {
                    return getVerticalSlider();
                } else if (name.equals("VerticalSplitArea")) {
                    return getVerticalSplitArea();
                } else if (name.equals("VerticalStack")) {
                    return getVerticalStack();
                }
                break;

//            case 'W':
//                if (name.equals("WrapArea")) {
//                    return getWrapArea();
//                } else if (name.equals("WrapGrid")) {
//                    return getWrapGrid();
//                }
//                break;
            }
        }

        return null;
    }

    public static BlankItem getBlankItem() {
        return new BlankItem();
    }

    public static ButtonCore getButtonCore() {
        return new ButtonCore();
    }

    public static ButtonToggle getButtonToggle() {
        return new ButtonToggle();
    }

    public static CheckBox getCheckBox() {
        return new CheckBox();
    }

    public static ComboBox getComboBox() {
        return new ComboBox();
    }

    public static ComboBoxDropDown getComboBoxDropDown() {
        return new ComboBoxDropDown();
    }

    //TODO
    public static ContextMenu getContextMenu(CoreWindow handler) {
        return new ContextMenu(handler);
    }

    //???
    public static CustomShape getCustomShape() {
        return new CustomShape();
    }

    //???
    public static Ellipse getEllipse() {
        return new Ellipse();
    }

    //TODO
    public static FileSystemEntry getFileSystemEntry(FileSystemEntryType type, String text) {
        return new FileSystemEntry(type, text);
    }

    //TODO
    public static FloatItem getFloatItem(CoreWindow handler) {
        return new FloatItem(handler);
    }

    public static Frame getFrame() {
        return new Frame();
    }

    public static FreeArea getFreeArea() {
        return new FreeArea();
    }

    public static Graph getGraph() {
        return new Graph();
    }

    public static Grid getGrid() {
        return new Grid(1, 1); //assume, that I'll set row/colCount later
    }

    public static HorizontalScrollBar getHorizontalScrollBar() {
        return new HorizontalScrollBar();
    }

    public static HorizontalSlider getHorizontalSlider() {
        return new HorizontalSlider();
    }

    public static HorizontalSplitArea getHorizontalSplitArea() {
        return new HorizontalSplitArea();
    }

    public static HorizontalStack getHorizontalStack() {
        return new HorizontalStack();
    }

    public static ImageItem getImageItem() {
        return new ImageItem();
    }

    public static Indicator getIndicator() {
        return new Indicator();
    }

    //TODO
    public static InputBox getInputBox(String title, String actionName, String defaultText) {
        return new InputBox(title, actionName, defaultText);
    }

    //TODO
    public static InputDialog getInputDialog(String title, String actionName, String defaultText) {
        return new InputDialog(title, actionName, defaultText);
    }

    public static Label getLabel() {
        return new Label();
    }

    public static LinesContainer getLinesContainer() {
        return new LinesContainer();
    }

    public static ListArea getListArea() {
        return new ListArea();
    }

    public static ListBox getListBox() {
        return new ListBox();
    }

    public static LoadingScreen getLoadingScreen() {
        return new LoadingScreen();
    }

    public static MenuItem getMenuItem() {
        return new MenuItem();
    }

    public static MessageBox getMessageBox() {
        return new MessageBox();
    }

    public static MessageItem getMessageItem() {
        return new MessageItem();
    }

    //TODO
    public static OpenEntryBox getOpenEntryBox(String title, FileSystemEntryType entryType, OpenDialogType dialogType) {
        return new OpenEntryBox(title, entryType, dialogType);
    }

    //TODO
    public static OpenEntryDialog getOpenEntryDialog(String title, FileSystemEntryType entryType, OpenDialogType dialogType) {
        return new OpenEntryDialog(title, entryType, dialogType);
    }

    public static PasswordLine getPasswordLine() {
        return new PasswordLine();
    }

    public static PointsContainer getPointsContainer() {
        return new PointsContainer();
    }

    //TODO
    public static PopUpMessage getPopUpMessage(String message) {
        return new PopUpMessage(message);
    }

    public static ProgressBar getProgressBar() {
        return new ProgressBar();
    }

    public static RadioButton getRadioButton() {
        return new RadioButton();
    }

    public static Rectangle getRectangle() {
        return new Rectangle();
    }

    public static ResizableItem getResizableItem() {
        return new ResizableItem();
    }

    public static ScrollHandler getScrollHandler() {
        return new ScrollHandler();
    }

    //TODO
    public static SelectionItem getSelectionItem(InterfaceBaseItem content) {
        return new SelectionItem(content);
    }

    //TODO
    public static SideArea getSideArea(CoreWindow handler, Side attachSide) {
        return new SideArea(handler, attachSide);
    }

    public static SpinItem getSpinItem() {
        return new SpinItem();
    }

    //TODO
    public static SplitHolder getSplitHolder(Orientation orientation) {
        return new SplitHolder(orientation);
    }

    public static Style getStyle() {
        return Style.getDefaultCommonStyle();
    }

    public static Tab getTab() {
        return new Tab();
    }

    public static TabView getTabView() {
        return new TabView();
    }

    public static TextArea getTextArea() {
        return new TextArea();
    }

    public static TextEdit getTextEdit() {
        return new TextEdit();
    }

    public static TextView getTextView() {
        return new TextView();
    }

    public static TitleBar getTitleBar() {
        return new TitleBar();
    }

    //TODO
    public static TreeItem getTreeItem(TreeItemType type) {
        return new TreeItem(type);
    }

    public static TreeView getTreeView() {
        return new TreeView();
    }

    public static Triangle getTriangle() {
        return new Triangle();
    }

    public static VerticalScrollBar getVerticalScrollBar() {
        return new VerticalScrollBar();
    }

    public static VerticalSlider getVerticalSlider() {
        return new VerticalSlider();
    }

    public static VerticalSplitArea getVerticalSplitArea() {
        return new VerticalSplitArea();
    }

    public static VerticalStack getVerticalStack() {
        return new VerticalStack();
    }

    //TODO
    public static WrapArea getWrapArea(int cellWidth, int cellHeight, Orientation orientation) {
        return new WrapArea(cellWidth, cellHeight, orientation);
    }

    //TODO
    public static WrapGrid getWrapGrid(int cellWidth, int cellHeight, Orientation orientation) {
        return new WrapGrid(cellWidth, cellHeight, orientation);
    }

}

File: sandbox/View/json/ItemsFactory.cs
using SpaceVIL;
using SpaceVIL.Core;
using SpaceVIL.Decorations;

namespace View.json {
    public class ItemsFactory {
        public static object GetObject(string name)
        {
            switch (name[0]) {
                case 'B':
                {
                    if (name.Equals("BlankItem"))
                    {
                        return GetBlankItem();
                    }
                    else if (name.Equals("ButtonCore"))
                    {
                        return GetButtonCore();
                    }
                    else if (name.Equals("ButtonToggle"))
                    {
                        return GetButtonToggle();
                    }
                    break;
                }
                case 'C':
                {
                    if (name.Equals("CheckBox"))
                    {
                        return GetCheckBox();
                    }
                    else if (name.Equals("ComboBox"))
                    {
                        return GetComboBox();
                    }
                    else if (name.Equals("ComboBoxDropDown"))
                    {
                        return GetComboBoxDropDown();
                    }
                    else if (name.Equals("CustomShape"))
                    {
                        return GetCustomShape();
                    }
                    break;
                }
                case 'E':
                {
                    if (name.Equals("Ellipse"))
                    {
                        return GetEllipse();
                    }
                    break;
                }
                case 'F':
                {
                    if (name.Equals("Frame"))
                    {
                        return GetFrame();
                    }
                    else if (name.Equals("FreeArea"))
                    {
                        return GetFreeArea();
                    }
                    break;
                }
                case 'G':
                {
                    if (name.Equals("Graph"))
                    {
                        return GetGraph();
                    }
                    else if (name.Equals("Grid"))
                    {
                        return GetGrid();
                    }
                    break;
                }
                case 'H':
                {
                    if (name.Equals("HorizontalScrollBar"))
                    {
                        return GetHorizontalScrollBar();
                    }
                    else if (name.Equals("HorizontalSlider"))
                    {
                        return GetHorizontalSlider();
                    }
                    else if (name.Equals("HorizontalSplitArea"))
                    {
                        return GetHorizontalSplitArea();
                    }
                    else if (name.Equals("HorizontalStack"))
                    {
                        return GetHorizontalStack();
                    }
                    break;
                }
                case 'I':
                {
                    if (name.Equals("ImageItem"))
                    {
                        return GetImageItem();
                    }
                    else if (name.Equals("Indicator"))
                    {
                        return GetIndicator();
                    }
                    break;
                }
                case 'L':
                {
                    if (name.Equals("Label"))
                    {
                        return GetLabel();
                    }
                    else if (name.Equals("LinesContainer"))
                    {
                        return GetLinesContainer();
                    }
                    else if (name.Equals("ListArea"))
                    {
                        return GetListArea();
                    }
                    else if (name.Equals("ListBox"))
                    {
                        return GetListBox();
                    }
                    else if (name.Equals("LoadingScreen"))
                    {
                        return GetLoadingScreen();
                    }
                    break;
                }
                case 'M':
                {
                    if (name.Equals("MenuItem"))
                    {
                        return GetMenuItem();
                    }
                    else if (name.Equals("MessageBox"))
                    {
                        return GetMessageBox();
                    }
                    else if (name.Equals("MessageItem"))
                    {
                        return GetMessageItem();
                    }
                    break;
                }

                case 'P':
                {
                    if (name.Equals("PasswordLine"))
                    {
                        return GetPasswordLine();
                    }
                    else if (name.Equals("PointsContainer"))
                    {
                        return GetPointsContainer();
                    }
                    else if (name.Equals("ProgressBar"))
                    {
                        return GetProgressBar();
                    }
                    break;
                }
                case 'R':
                {
                    if (name.Equals("RadioButton"))
                    {
                        return GetRadioButton();
                    }
                    else if (name.Equals("Rectangle"))
                    {
                        return GetRectangle();
                    }
                    else if (name.Equals("ResizableItem"))
                    {
                        return GetResizableItem();
                    }
                    break;
                }
                case 'S':
                {
                    if (name.Equals("ScrollHandler"))
                    {
                        return GetScrollHandler();
                    }
                    else if (name.Equals("SpinItem"))
                    {
                        return GetSpinItem();
                    }
                    else if (name.Equals("Style") || name.Equals("InnerStyle")) //???
                    {
                        return GetStyle();
                    }
                    break;
                }
                case 'T':
                {
                    if (name.Equals("Tab"))
                    {
                        return GetTab();
                    }
                    else if (name.Equals("TabView"))
                    {
                        return GetTabView();
                    }
                    else if (name.Equals("TextArea"))
                    {
                        return GetTextArea();
                    }
                    else if (name.Equals("TextEdit"))
                    {
                        return GetTextEdit();
                    }
                    else if (name.Equals("TextView"))
                    {
                        return GetTextView();
                    }
                    else if (name.Equals("TitleBar"))
                    {
                        return GetTitleBar();
                    }
                    else if (name.Equals("TreeView"))
                    {
                        return GetTreeView();
                    }
                    else if (name.Equals("Triangle"))
                    {
                        return GetTriangle();
                    }
                    break;
                }
                case 'V':
                {
                    if (name.Equals("VerticalScrollBar"))
                    {
                        return GetVerticalScrollBar();
                    }
                    else if (name.Equals("VerticalSlider"))
                    {
                        return GetVerticalSlider();
                    }
                    else if (name.Equals("VerticalSplitArea"))
                    {
                        return GetVerticalSplitArea();
                    }
                    else if (name.Equals("VerticalStack"))
                    {
                        return GetVerticalStack();
                    }
                    break;
                }
            }
            return null;
        }

        public static BlankItem GetBlankItem()
        {
            return new BlankItem();
        }
    
        public static ButtonCore GetButtonCore()
        {
            return new ButtonCore();
        }
    
        public static ButtonToggle GetButtonToggle()
        {
            return new ButtonToggle();
        }
    
        public static CheckBox GetCheckBox()
        {
            return new CheckBox();
        }
    
        public static ComboBox GetComboBox()
        {
            return new ComboBox();
        }
    
        public static ComboBoxDropDown GetComboBoxDropDown()
        {
            return new ComboBoxDropDown();
        }
    
        //TODO
        public static ContextMenu GetContextMenu(CoreWindow handler)
        {
            return new ContextMenu(handler);
        }
    
        //???
        public static CustomShape GetCustomShape()
        {
            return new CustomShape();
        }
    
        //???
        public static Ellipse GetEllipse()
        {
            return new Ellipse();
        }
    
        //TODO
        public static FileSystemEntry GetFileSystemEntry(FileSystemEntryType type, string text)
        {
            return new FileSystemEntry(type, text);
        }
    
        //TODO
        public static FloatItem GetFloatItem(CoreWindow handler)
        {
            return new FloatItem(handler);
        }
    
        public static Frame GetFrame()
        {
            return new Frame();
        }
    
        public static FreeArea GetFreeArea()
        {
            return new FreeArea();
        }
    
        public static Graph GetGraph()
        {
            return new Graph();
        }
    
        public static Grid GetGrid()
        {
            return new Grid(1, 1); //assume, that I'll set row/colCount later
        }
    
        public static HorizontalScrollBar GetHorizontalScrollBar()
        {
            return new HorizontalScrollBar();
        }
    
        public static HorizontalSlider GetHorizontalSlider()
        {
            return new HorizontalSlider();
        }
    
        public static HorizontalSplitArea GetHorizontalSplitArea()
        {
            return new HorizontalSplitArea();
        }
    
        public static HorizontalStack GetHorizontalStack()
        {
            return new HorizontalStack();
        }
    
        public static ImageItem GetImageItem()
        {
            return new ImageItem();
        }
    
        public static Indicator GetIndicator()
        {
            return new Indicator();
        }
    
        //TODO
        public static InputBox GetInputBox(string title, string actionName, string defaultText)
        {
            return new InputBox(title, actionName, defaultText);
        }
    
        //TODO
        public static InputDialog GetInputDialog(string title, string actionName, string defaultText)
        {
            return new InputDialog(title, actionName, defaultText);
        }
    
        public static Label GetLabel()
        {
            return new Label();
        }
    
        public static LinesContainer GetLinesContainer()
        {
            return new LinesContainer();
        }
    
        public static ListArea GetListArea()
        {
            return new ListArea();
        }
    
        public static ListBox GetListBox()
        {
            return new ListBox();
        }
    
        public static LoadingScreen GetLoadingScreen()
        {
            return new LoadingScreen();
        }
    
        public static MenuItem GetMenuItem()
        {
            return new MenuItem();
        }
    
        public static MessageBox GetMessageBox()
        {
            return new MessageBox();
        }
    
        public static MessageItem GetMessageItem()
        {
            return new MessageItem();
        }
    
        //TODO
        public static OpenEntryBox GetOpenEntryBox(string title, FileSystemEntryType entryType, OpenDialogType dialogType)
        {
            return new OpenEntryBox(title, entryType, dialogType);
        }
    
        //TODO
        public static OpenEntryDialog GetOpenEntryDialog(string title, FileSystemEntryType entryType, OpenDialogType dialogType)
        {
            return new OpenEntryDialog(title, entryType, dialogType);
        }
    
        public static PasswordLine GetPasswordLine()
        {
            return new PasswordLine();
        }
    
        public static PointsContainer GetPointsContainer()
        {
            return new PointsContainer();
        }
    
        //TODO
        public static PopUpMessage GetPopUpMessage(string message)
        {
            return new PopUpMessage(message);
        }
    
        public static ProgressBar GetProgressBar()
        {
            return new ProgressBar();
        }
    
        public static RadioButton GetRadioButton()
        {
            return new RadioButton();
        }
    
        public static Rectangle GetRectangle()
        {
            return new Rectangle();
        }
    
        public static ResizableItem GetResizableItem()
        {
            return new ResizableItem();
        }
    
        public static ScrollHandler GetScrollHandler()
        {
            return new ScrollHandler();
        }
    
        //TODO
        public static SelectionItem GetSelectionItem(IBaseItem content)
        {
            return new SelectionItem(content);
        }
    
        //TODO
        public static SideArea GetSideArea(CoreWindow handler, Side attachSide)
        {
            return new SideArea(handler, attachSide);
        }
    
        public static SpinItem GetSpinItem()
        {
            return new SpinItem();
        }
    
        //TODO
        public static SplitHolder GetSplitHolder(Orientation orientation)
        {
            return new SplitHolder(orientation);
        }
    
        public static Style GetStyle()
        {
            return Style.GetDefaultCommonStyle();
        }
    
        public static Tab GetTab()
        {
            return new Tab();
        }
    
        public static TabView GetTabView()
        {
            return new TabView();
        }
    
        public static TextArea GetTextArea()
        {
            return new TextArea();
        }
    
        public static TextEdit GetTextEdit()
        {
            return new TextEdit();
        }
    
        public static TextView GetTextView()
        {
            return new TextView();
        }
    
        public static TitleBar GetTitleBar()
        {
            return new TitleBar();
        }
    
        //TODO
        public static TreeItem GetTreeItem(TreeItemType type)
        {
            return new TreeItem(type);
        }
    
        public static TreeView GetTreeView()
        {
            return new TreeView();
        }
    
        public static Triangle GetTriangle()
        {
            return new Triangle();
        }
    
        public static VerticalScrollBar GetVerticalScrollBar()
        {
            return new VerticalScrollBar();
        }
    
        public static VerticalSlider GetVerticalSlider()
        {
            return new VerticalSlider();
        }
    
        public static VerticalSplitArea GetVerticalSplitArea()
        {
            return new VerticalSplitArea();
        }
    
        public static VerticalStack GetVerticalStack()
        {
            return new VerticalStack();
        }
    
        //TODO
        public static WrapArea GetWrapArea(int cellWidth, int cellHeight, Orientation orientation)
        {
            return new WrapArea(cellWidth, cellHeight, orientation);
        }
    
        //TODO
        public static WrapGrid GetWrapGrid(int cellWidth, int cellHeight, Orientation orientation)
        {
            return new WrapGrid(cellWidth, cellHeight, orientation);
        }
    }
}